### ISSUE TITLE
introduce root2json for common workspaces?

### ISSUE THREAD

**Issue by lukasheinrich** [diana-hep/pyhf#542]
# Description

It seems one of the main avenunes we generate pyhf json is using `PrintXML()` on a `Measurement` object. In that light a `pyhf root2json /path/to/workspace.root` seems nice. On the other hand, this will require a full blown ROOT installation (as opposed to only uproot in xml2json)

**Comment by matthewfeickert** [comment]
I would agree that it seems nice, but I'm a hesitant to accept ROOT as an optional dependency. I would prefer to keep pyhf having the "benefit" of no ROOT involved at all through use of uproot.

I might just be overly grumpy on ROOT here though.

**Comment by kratsg** [comment]
I'm really hesitant here. Is it not possible to say... perhaps write a custom parser/decoder with `uproot` to handle the class here? @jpivarski probably can comment. It would require some concerted effort on the `uproot` side, but ...

**Comment by jpivarski** [comment]
My only brush with RooFit objects was scikit-hep/uproot#169. In principle, uproot should be able to read these objects, but they seem to be touching some aspects of object deserialization that I haven't implemented. (There are a few `_raise_notimplemented` calls in the [TStreamer-to-Python-class generator](https://github.com/scikit-hep/uproot/blob/df8467f3bb5f6c17af556189d464e1be7fd29d74/uproot/rootio.py#L670-L848).)

It's not clear whether this would be a shallow or a deep rabbit hole, but closing these `_raise_notimplemented` stubs would benefit more data types than just the RooFit ones. I don't have the time to work on that directly, but I do have the time to guide someone through, if they take it up as a project. But as I said, I can't say whether it would be a short project or a long project, even given a goal (such as, "fully read _this_ ROOT file"). It might be a matter of solving one new TStreamer type or several.

At least these objects are not inside of TTrees, which affects their deserialization in complex ways (sometimes they have byte count headers, sometimes not; sometimes they have TObject headers, sometimes not, depending on their nesting level...). Objects outside of TTrees, such as those involved in a RooWorkspace, are more regular: if you've solved it for the outer level of nesting, you've solved it for all levels of nesting.
